src/detection/DepthImageObstacleDetector.cc

```diff
@@ -284,25 +285,47 @@ int DepthImageObstacleDetector::extract_blobs()
 
                 blob_to_obstacle[label] = num_obstacles++;
                 obstacles[blob_to_obstacle[label]].id = label;
-                obstacles[blob_to_obstacle[label]].center.x = DBL_MAX;
+                obstacles[blob_to_obstacle[label]].bounding_box.tlc.y = i;
             }
 
             Obstacle *o = &obstacles[blob_to_obstacle[label]];
             o->center += glm::dvec3(0, i, j);
-            o->center.x = (depth_frame[row_offset + j] < o->center.x) ?
-                depth_frame[row_offset + j] : o->center.x;
+
+            if (depth_frame[row_offset + j] < o->bounding_box.tlc.x)
+                o->bounding_box.tlc.x = depth_frame[row_offset + j];
+
+            if (depth_frame[row_offset + j] > o->bounding_box.brf.x)
+                o->bounding_box.brf.x = depth_frame[row_offset + j];
+
+            if (i > o->bounding_box.brf.y)
+                o->bounding_box.brf.y = i;
+
+            if (j > o->bounding_box.brf.z)
+                o->bounding_box.brf.z = j;
+
+            if (j < o->bounding_box.tlc.z)
+                o->bounding_box.tlc.z = j;
         }
     }
 
     this->obstacles.resize(num_obstacles);
     for (Obstacle &o : obstacles) {
-        o.center.x *= this->scale;
+        o.bounding_box.brf.x *= this->scale;
+        o.bounding_box.tlc.x *= this->scale;
+
+        o.center.x = o.bounding_box.tlc.x;
```

Comment:
True, but since the radius of the center doesn't match position of the obstacle at that point (the closest point isn't actually in the center), we risk to leave part of the object out of the bounding box (and therefore isn't a very good bounding box anymore :-P) if we use the center x, y coords to calculate the radius on the corner to make a plane going through there.

To calculate the appropriated radius like you said, we need to keep track of real coordinate of the closest point. This can be done for the Obstacle detector, we just need to keep track of more stuff while calculating the blobs, but I don't think it will be possible for the polar hist, we make endup leaving part of the object outside the bounding box if it happens that the closest point is near the edges of the slice.